dapps/assets/src/components/digitaltwins/AddDigitalTwin.ts

```diff
@@ -100,62 +128,47 @@ class AddDigitalTwinComponent extends mixins(EvanComponent) {
   }
 
   /**
-   * Returns localized string from template or general if not found.
+   * Adds e new entry to the twin selection list.
    *
-   * @param template
+   * @param template - DigitalTwinTemplate
+   * @param twinKey - unique key per template
    */
-  getLocalizedTemplateEntry(template: DigitalTwinTemplate, entry: string): string {
-    const lang = window.localStorage.getItem('evan-language');
+  _addToPresetTemplates(template: DigitalTwinTemplate, key: string) {
+    this.twinTemplates[key] = template;
+    this.presetTemplates = this._getTemplateSelectOptions();
+  }
 
-    return template.description?.i18n[lang]?.[entry] || template.description?.[entry] || '';
+  _getTemplateSelectOptions() {
+    return Object.keys(this.twinTemplates).map(twinKey => {
+      return {
+        value: twinKey,
+        label: this._getLocalizedTemplateEntry(this.twinTemplates[twinKey], 'name'),
+      };
+    });
   }
 
   /**
-   * Set empty fields from template and update current template by uploaded file.
+   * Returns localized string from template or general if not found.
    *
-   * @param files
+   * @param template
    */
-  async handleFileUpload (files: UIContainerFile[]) {
-    // reset when file was deleted
-    if (!files[0]) {
-      this.template = null;
+  _getLocalizedTemplateEntry(template: DigitalTwinTemplate, entry: string): string {
+    const lang = this.$i18n.locale() || window.localStorage.getItem('evan-language');
 
-      return;
-    }
-
-    this.template = await this._blobToObj(files[0].blob) as DigitalTwinTemplate;
+    return template.description?.i18n[lang]?.[entry] || template.description?.[entry] || '';
+  }
 
+  /**
+   * Set name and description from twin template when not set before.
+   */
+  _setDefaults() {
```

Comment:
What does the underscore mean? If private, just set it to private instead. Underscore is legacy from old JS :)
Same applies to the other methods.